src/shared/codelens/codeLensUtils.ts

```diff
@@ -139,3 +147,36 @@ export function getMetricDatum({command, isDebug, runtime}: {
         }
     }
 }
+
+async function getAssociatedSamTemplate(
+    documentUri: vscode.Uri,
+    workspaceFolderUri: vscode.Uri
+): Promise<vscode.Uri> {
+
+    // Get Template files in Workspace
+    const templatesAsync: AsyncIterableIterator<vscode.Uri> = detectLocalTemplates({
+        workspaceUris: [workspaceFolderUri]
+    })
+
+    // See which templates (if any) are in a folder, or parent folder, of the document of interest
+    const templates = await toArrayAsync(templatesAsync)
+    const candidateTemplates = templates
+        .filter(template => {
+            const folder = dirname(template.fsPath)
+
+            return documentUri.fsPath.indexOf(folder) === 0
```

Comment:
Nit: `documentUri.fsPath.contains(folder)`

Reply:
Also, do we need to normalize `folder` to lowercase first?

Reply:
"Contains folder" can cause a false positive, "Starts with folder" seems more correct.
All of the folder casing should be pure in the code flows, so it seems like performing a case insensitive comparison would give incorrect results, which would be risky running case sensitive OSes.

Given `/folder/aaa/src/code.js` :
* `/folder/aaa/template.yaml` should match
* `/folder/AAA/template.yaml` should not match

Reply:
* Ah, gotcha. In that case, use `documentUri.fsPath.startsWith`.
* `/folder/AAA/template.yaml` should match on Windows, but not Posix.

Reply:
Will we if we never fiddle with path casing in our code?